src/core/sock/sockinfo.cpp

```diff
@@ -425,6 +427,19 @@ int sockinfo::setsockopt(int __level, int __optname, const void *__optval, sockl
                           "optval == NULL");
             }
             break;
+
+        case SO_REUSEADDR:
+            m_reuseaddr = *(bool *)__optval;
```

Comment:
[suggestion]
At least Linux kernel implementation treats optval as "int boolean". I believe standard also doesn't describe 1 byte boolean. In case application passes 0x100 as the argument, current code will convert it to false.

We can use the same way as the kernel:
1. Globally check that __optlen is at least sizeof(int)
2. Convert pointer to (int *)